src/components/UI/ListItems/PlioListItem.vue

```diff
@@ -0,0 +1,186 @@
+      <!-- plio title -->
+      <div>
+        <p class="text-xl font-bold">{{ title }}</p>
+      </div>
+
+      <!-- plio link -->
+      <div>
+        <URL
+          :link="plioLink"
+          :urlStyleClass="urlStyleClass"
+          :urlCopyButtonClass="urlCopyButtonClass"
+        ></URL>
+      </div>
+
+      <!-- action buttons -->
+      <div class="flex flex-row justify-start gap-3">
+        <!-- play button -->
+        <icon-button
+          :titleConfig="playButtonTitleConfig"
+          :iconConfig="playButtonIconConfig"
+          :buttonClass="playButtonClass"
+          @click="playPlio"
+          :isDisabled="!isPublished"
+          v-tooltip="playButtonTooltip"
+        ></icon-button>
+
+        <!-- edit button -->
+        <icon-button
+          :titleConfig="editButtonTitleConfig"
+          :iconConfig="editButtonIconConfig"
+          :buttonClass="editButtonClass"
+          @click="editPlio"
+        ></icon-button>
+
+        <!-- duplicate button -->
+        <icon-button
+          :titleConfig="duplicateButtonTitleConfig"
+          :iconConfig="duplicateButtonIconConfig"
+          :buttonClass="duplicateButtonClass"
+          @click="duplicatePlio"
+        ></icon-button>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import PlioAPIService from "@/services/API/Plio.js";
+import URL from "@/components/UI/Text/URL.vue";
+import IconButton from "@/components/UI/Buttons/IconButton.vue";
+import SimpleBadge from "@/components/UI/Badges/SimpleBadge.vue";
+import PlioListItemSkeleton from "@/components/UI/Skeletons/PlioListItemSkeleton.vue";
+
+export default {
+  name: "PlioThumbnail",
+  props: ["plioId"],
+  components: {
+    URL,
+    IconButton,
+    SimpleBadge,
+    PlioListItemSkeleton,
+  },
+
+  data() {
+    return {
+      plioDetails: {},
+      playButtonTitleConfig: {
+        value: "Play",
+        class: "p-2 text-primary font-semibold",
+      },
+      playButtonIconConfig: {
+        enabled: false,
+        iconName: "",
+        iconClass: "",
+      },
+      playButtonClass: "bg-gray-100 hover:bg-gray-200 rounded-md shadow-md h-10",
+
+      editButtonTitleConfig: {
+        value: "Edit",
+        class: "p-2 text-primary font-semibold",
+      },
+      editButtonIconConfig: {
+        enabled: false,
+        iconName: "",
+        iconClass: "",
+      },
+      editButtonClass: "bg-gray-100 hover:bg-gray-200 rounded-md shadow-md h-10",
+
+      duplicateButtonTitleConfig: {
+        value: "Duplicate",
+        class: "p-2 text-primary font-semibold",
+      },
+      duplicateButtonIconConfig: {
+        enabled: false,
+        iconName: "",
+        iconClass: "",
+      },
+      duplicateButtonClass: "bg-gray-100 hover:bg-gray-200 rounded-md shadow-md h-10",
+      urlStyleClass: "text-xs font-bold text-yellow-600",
+      urlCopyButtonClass: "text-yellow-600",
+      isDataLoaded: false,
+    };
+  },
+
+  async created() {
+    this.isDataLoaded = false;
```

Comment:
why is this needed?

Reply:
Added it so that I can show the skeleton until a plio's data is loaded
- I think it's better to do it using vuex ka startLoading and stopLoading